src/model/mmodule.nit

```diff
@@ -104,6 +112,22 @@ class MModule
 		end
 	end
 
+	# The namespace used for public entities.
+	#
+	# Public entities use only the project as a namespace.
+	# Private entities use the `full_name` (i.e. "project::module")
+	#
+	# This method is used by entities to implement their `full_name`.
+	fun namespace_for(v: MVisibility): String do
+		if v <= private_visibility then return full_name
```

Comment:
By reading the synopsis one can think that this method do a noop for private entities.